Template improvements

DerAndereAndi opened the pull request:
Fix #2018

- Update go-e firmware requirements
- Change cfos devices not to use modbus config schema
- Add support for `onIdentify` params of vehicles
- Add support for `resetOnDisconnect` for loadpoints
- Add support for defining device specific modbus port, baudrate and comset defaults
- Handle fatal error when `evcc.yaml` exists and the process doesn't have write permissions
- Some bug fixes

DerAndereAndi committed:
Update go-e firmware requirements

DerAndereAndi committed:
Add support for onIdentify params of vehicles

DerAndereAndi committed:
Add support for resetOnDisconnect on loadpoints

DerAndereAndi commented:
@andig wenn du magst kannst schon mal drüber schauen

ansonsten was sollte noch dazu? @naltatis @premultiply @andig ?

DerAndereAndi committed:
Fix display of nil exampleValue

DerAndereAndi committed:
Update README

DerAndereAndi committed:
Fix invalid assignment

DerAndereAndi committed:
Add default value for valuetypechargemodes

DerAndereAndi committed:
Add support to predefine modbus port

DerAndereAndi committed:
Set modbus defaults for some devices

DerAndereAndi committed:
Fix cfos meter not being provided as charge meter

DerAndereAndi committed:
Fix build

DerAndereAndi committed:
Fix templates using modbus creating invalid yaml

DerAndereAndi committed:
Fix linter warning

DerAndereAndi committed:
A fix for modbus

the modbus key should not be removed, otherwise we can't render the values properly

DerAndereAndi committed:
Some code cleanup

DerAndereAndi committed:
Fix linter warning

DerAndereAndi committed:
Check for write permissions

DerAndereAndi committed:
Change cfos devices not to use modbus config

DerAndereAndi committed:
Improve file permissions handling

DerAndereAndi committed:
cfos meter setup needs port and id

andig commented on cmd/configure/survey.go:
```diff
@@ -137,6 +139,22 @@ func (c *CmdConfigure) askValue(q question) string {
+	if q.valueType == templates.ParamValueTypeChargeModes {
+		chargingModes := []string{string(api.ModeOff), string(api.ModeNow), string(api.ModeMinPV), string(api.ModePV)}
+		chargeModes := []string{
+			c.localizedString("ChargeModeOff", nil),
```
macht die Übersetzung Sinn wenn ich im Gui was Anderes sehe? Oder ist das auch schon mehrsprachig?

DerAndereAndi replied:
Das relevante da sehe ich eher im Erklärtext zu den Modi. Schaus dir einfach mal an. Übersetzt ist da auch nur "Aus" und "Sofort". Und das kann man im englischen lassen und dann in die Erklärung reinbringen.

DerAndereAndi replied:
<img width="442" alt="image" src="https://user-images.githubusercontent.com/42185/145707821-bfb98dac-b360-4c10-85f8-695c13a659bd.png">

Das sehe ich in der UI. Müsste also noch aus "Aus" dann "Stop" machen.

andig commented on cmd/configure/texts.go:
```diff
@@ -81,6 +81,53 @@ func (c *CmdConfigure) userFriendlyTexts(param templates.Param) templates.Param
 		if result.Help.String(c.lang) == "" {
 			result.Help.SetString(c.lang, c.localizedString("UserFriendly_Vin_Help", nil))
 		}
+	case "cache":
```
Wäre es nicht einfacher, das ganze Case Statement durch eine Map zu ersetzen?

DerAndereAndi replied:
joah, könnte man machen ;)

andig commented on templates/definition/charger/abl.yaml:
```diff
@@ -5,6 +5,8 @@ requirements:
 - name: modbus
   choice: ["rs485"]
+  baudrate: 38400
+  comset: "8E1"
```
braucht keine Anführungszeichen mehr, sauberer ists aber

DerAndereAndi replied:
bedeutet?

andig commented on templates/docs/meter/cfos-meter-charge.yaml:
```diff
@@ -1,4 +1,7 @@
 type: template
 template: cfos-meter
 description: cFos PowerBrain
-usage: charger
```
warum hat unser render Test da nicht gefailt? Fehlt was in der Fehlerbehandlung?

DerAndereAndi replied:
Das ist die generierte Doc die eh noch nicht benutzt wird und Arbeit braucht, das ist dann aber abhängig von der Integration in die Doku

andig replied:
Mir gehts darum, dass bei falscher usage egtl. ein Fehler vom Charger kommen müsste wenn der getestet wird. Das ist hier aber nicht der Fall- fehlt vmtl. im Charger Code und sollte abgefangen werden.

andig replied:
Anders gesagt: irgendwo fangen wir hier einen Fehler nicht ab. Im cfos Charger ist kein usage drin. Mit Nutzung der Doku hat das m.E. nix zu tun.

DerAndereAndi replied:
Hab mal checks dafür eingebaut

andig commented on util/templates/template_modbus.go:
```diff
@@ -12,6 +12,49 @@ import (
+			}
+			t.Params = append(t.Params, Param{Name: ModbusParamNameHost, ValueType: ParamValueTypeString})
+		case ModbusParamNamePort:
+			if values[ParamModbus] != ModbusKeyTCPIP && values[ParamModbus] != ModbusKeyRS485TCPIP {
```
if v := values[ParamModbus]; v != ...

andig commented on util/templates/template_modbus.go:
```diff
@@ -12,6 +12,49 @@ import (
+				continue
+			}
+			t.Params = append(t.Params, Param{Name: ModbusParamNamePort, ValueType: ParamValueTypeNumber})
+		case ModbusParamNameDevice:
```
Die sehen alle sehr ähnlich aus- durch 2 cases und 1 oder 2 Maps ersetzen?

DerAndereAndi committed:
Review changes

DerAndereAndi commented:
@andig die `onIdentify` parameter sind hiermit flach eingepflegt. Man könnte diese noch anstatt `targetSoC` als `onIdentifyTargetSoC` definieren, damit evtl. zukünftige  Konflikte nicht auftreten können. Aber denke das ist hier nicht unbedingt notwendig. Oder was meinst du?

DerAndereAndi committed:
Add modbus and usage choice validation

- Validates on doc generation
- Validates on template loading
- Can be extended with more checks

DerAndereAndi committed:
Add support for overwriting modbus default id

DerAndereAndi committed:
Migrate some templates to use modbus param

andig commented:
Fix https://github.com/evcc-io/evcc/issues/2018

DerAndereAndi committed:
Add fritzdect meter template

DerAndereAndi committed:
Update cfos meter template
